data_structures/binary_tree/symmetric_tree.py

```diff
@@ -0,0 +1,95 @@
+"""
+Given the root of a binary tree, check whether it is a mirror of itself
+(i.e., symmetric around its center).
+
+Leetcode reference: https://leetcode.com/problems/symmetric-tree/
+"""
+
+
+class Node:
+    """
+    A Node has data variable and pointers to Nodes to its left and right.
+    """
+
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.left: Node | None = None
+        self.right: Node | None = None
+
+
+class SymmetricTree:
+    def is_symmetric_tree(self, tree: Node) -> bool:
+        r"""
+        Test cases for is_symmetric_tree function.
+
+        The below tree looks like this
+               1
+             /   \
+            2      2
+           / \    / \
+          3   4   4  3
+
+        >>> tree = Node(1)
+        >>> tree.left = Node(2)
+        >>> tree.right = Node(2)
+        >>> tree.left.left = Node(3)
+        >>> tree.left.right = Node(4)
+        >>> tree.right.left = Node(4)
+        >>> tree.right.right = Node(3)
+
+        >>> SymmetricTree().is_symmetric_tree(tree)
+        True
```

Comment:
```suggestion
        >>> SymmetricTree().is_symmetric_tree(make_symmetric_tree())
        True
        >>> SymmetricTree().is_symmetric_tree(make_asymmetric_tree())
        False
```

Reply:
@cclauss  I have made the requested changes, please check